scripts/cangen/cangen/config.py

```diff
@@ -0,0 +1,47 @@
+import yaml
+
+
+class Bus:
+    def __init__(self, bus: dict):
+        self.bus = bus
+
+        self.dbc_file_path: str = self.bus.pop("dbcFile")
+        self.bus_name: str = self.bus.pop("busName").capitalize()
+
+        if not self.has_bus_been_consumed():
+            print(self.bus)
+            print(
+                f"{self.bus.keys()} field/s not expected in configuration file for bus {self.bus_name}."
+            )
+            raise ValueError
+
+    # Checks to see if all fields of inputted bus object have been used
+    def has_bus_been_consumed(self) -> bool:
+        return not self.bus
+
+
+class Config:
+    @staticmethod
+    def load_yaml(config_file_name: str) -> dict:
+        with open(config_file_name, "r") as file:
+            config = yaml.safe_load(file)
+        return config.pop("canGen")
+
+    def __init__(self, config_file_name: str):
```

Comment:
`__init__` should take a dictionary, not a filename

```python
def __init__(self config: dict):
   ...
```

init should not handle the file at all. Rather, the `load_yaml` method show be replaced with a `from_yaml` method with the form

```python
@staticmethod
def from_yaml(file_name: str) -> Config:
    with open(filename ...
        config = ...
    return Config(config.pop("canGen"))
```

This is a big difference. It means that we can create Configs from any dictionary, but provide a convenient method for creating one from known yaml format. You code in `can_generator.py` will now be `config = Config.from_yaml(...`

Other details
- Since we are in the Config class, it is redundant to have "config_" in the filename variable
- You will need `from __future__ import annotations` to use the `-> Config` signature